src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1295,13 +1295,17 @@ def aks_get_credentials(client, resource_group_name, name, admin=False,
                 pass
 
         # merge the new kubeconfig into the existing one
-        with tempfile.NamedTemporaryFile(mode='w+t') as additional_file:
+        fd, temp_path = tempfile.mkstemp()
+        additional_file = os.fdopen(fd, 'w+t')
+        try:
             additional_file.write(kubeconfig)
             additional_file.flush()
-            try:
-                merge_kubernetes_configurations(path, additional_file.name)
-            except yaml.YAMLError as ex:
-                logger.warning('Failed to merge credentials to kube config file: %s', ex)
+            merge_kubernetes_configurations(path, temp_path)
+        except yaml.YAMLError as ex:
+            logger.warning('Failed to merge credentials to kube config file: %s', ex)
+        finally:
+            additional_file.close()
+            os.remove(temp_path)
```

Comment:
I suggest you add a couple of tests for this logic. The related issue impacts quite a few users, so good to build up regression tests

Reply:
Also, a bit straightforward option is to pass the content to `merge_kubernetes_configurations` rather build up a temp file and gets read out again.  Of course this requires some refactor on old code, so up to you.